node/network/pov-distribution/src/lib.rs

```diff
@@ -289,7 +366,21 @@ async fn handle_fetch(
 				return Ok(());
 			}
 			Entry::Vacant(e) => {
-				e.insert(vec![response_sender]);
+				if let Some(relevant_validators) = determine_relevant_validators(
+					ctx,
+					relay_parent,
+					descriptor.para_id,
+				).await? {
+					let new_connection_request = validator_discovery::connect_to_validators(
+						ctx,
+						relay_parent,
+						relevant_validators.clone(),
+					).await?;
+
+					state.connection_requests.put(relay_parent, new_connection_request);
```

Comment:
I suppose we only need one connection request per `(relay_parent, para_id)` pair, in actuality. And even this is possibly overkill.

Reply:
agree